### PR TITLE
Plank: continue to sync on failure to list PJs in a build cluster

### LINKED ISSUE #15200: Plank crashes when unable to connect to a single build cluster.
**What happened**:
Plank crashes (i.e. fails to sync) when a build cluster in unavailable, inaccessible, or nonexistent - even if the build cluster does not have any jobs. 
```console
{
 insertId: "1jhyicjg3kaixgf"  
 jsonPayload: {
  component: "plank"   
  error: "error listing pods in cluster "agones": Get https://35.233.156.120/api/v1/namespaces/test-pods/pods?labelSelector=created-by-prow%3Dtrue: x509: certificate signed by unknown authority"   
  file: "prow/cmd/plank/main.go:144"   
  func: "main.main.func2"   
  level: "error"   
  msg: "Error syncing."   
 }
 labels: {…}  
 logName: "projects/oss-prow/logs/plank"  
 receiveTimestamp: "2019-11-08T20:35:49.483304919Z"  
 resource: {…}  
 severity: "ERROR"  
 timestamp: "2019-11-08T20:35:44Z"  
}
```

**What you expected to happen**:
Plank should log an error (and/or send an alert) regarding failure to reach the cluster and *continue* to create Pods for other *properly-configured* clusters.    

**How to reproduce it (as minimally and precisely as possible)**:
Add a build cluster w/ an **invalid** `server` endpoint (or CA or cert or key) to the `kubeconfig` file secret (in a Prow service cluster) and restart Plank.

**e.g.**
```yaml
apiVersion: v1
clusters:
  - cluster:
      certificate-authority-data: some-valid-ca
      server: https://some.invalid.endpoint
    name: some-valid-name
contexts:
  - context:
      cluster: some-valid-name
      user: some-valid-name
    name: some-valid-name
current-context: some-valid-name
kind: Config
preferences: {}
users:
  - name: some-valid-name
    user:
      token: some-valid-token
```

**Please provide links to example occurrences, if any**:
[oss-prow](https://pantheon.corp.google.com/logs/viewer?project=oss-prow&minLogLevel=0&expandAll=false&timestamp=2019-11-08T20%3A36%3A50.459000000Z&customFacets=jsonPayload.level%2CjsonPayload.component%2CjsonPayload.msg&limitCustomFacetWidth=false&interval=PT1H&resource=container%2Fcluster_name%2Fprow&scrollTimestamp=2019-11-08T20%3A35%3A56.000000000Z&dateRangeStart=2019-11-08T19%3A36%3A03.206Z&advancedFilter=resource.type%3D%22container%22%0Aresource.labels.project_id%3D%22oss-prow%22%0Aresource.labels.cluster_name%3D%22prow%22%0Aresource.labels.container_name%3D%22plank%22%0Aresource.labels.namespace_id%3D%22default%22%0Aresource.labels.instance_id%3D%228312772924294783708%22%0Aresource.labels.pod_id%3D%22plank-5cdb46fc4c-g522l%22%0Aresource.labels.zone%3D%22us-west1-a%22%0Atimestamp%3D%222019-11-08T20%3A35%3A44.000000000Z%22%0AinsertId%3D%221jhyicjg3kaixgf%22&dateRangeUnbound=forwardInTime)

/cc @BenTheElder 
/kind oncall-hotlist

### PR DESCRIPTION
**Pull request by clarketm**
We cannot allow an improperly configured, deleted, or inaccessible `build` cluster to take down Prow.

fix #15200 

/assign @BenTheElder @cjwagner

### CODE REVIEW

**Comment by BenTheElder** [comment]
thanks for the fix, this looks about right to me, but leaving to a prow OWNER to approve

**Comment by fejta** [comment]
> leaving to a prow OWNER to approve

That should include you

File: prow/plank/controller.go [hunk 1]
```diff
@@ -218,11 +220,12 @@ func (c *Controller) Sync() error {
 	for alias, client := range c.buildClients {
 		pods, err := client.List(metav1.ListOptions{LabelSelector: selector})
 		c.log.WithField("selector", selector).Debug("List Pods.")
-		if err != nil {
-			return fmt.Errorf("error listing pods in cluster %q: %v", alias, err)
-		}
-		for _, pod := range pods.Items {
-			pm[pod.ObjectMeta.Name] = pod
+		if err == nil {
```
[the remark this answers is not in the document]
Comment (clarketm) [reply prow/plank/controller.go]: Updated 👍

**Pull request merged**